VideoLocker/src/main/java/org/edx/mobile/player/PlayerFragment.java

```diff
@@ -1352,61 +1354,34 @@ private AccessibilityManager getAccessibilityManager() {
      */
     private boolean getTouchExploreEnabled() {
         boolean ret = false;
-        AccessibilityManager am = getAccessibilityManager();
-        if (am != null && am.isTouchExplorationEnabled()) {
-            ret = true;
-        }
-        return ret;
-    }
 
-    /**
-     * @param am The accessibility manager to unregister from
-     * @return True if successful
-     */
-    private boolean unregisterTouchExploreChangeListener(@NonNull AccessibilityManager am) {
-        boolean ret = false;
-        if (touchExplorationStateChangeListener == null) {
-            ret = true;
-        }
-        else if (am.removeTouchExplorationStateChangeListener(touchExplorationStateChangeListener)) {
-            touchExplorationStateChangeListener = null;
-            ret = true;
+        if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+            AccessibilityManager am = getAccessibilityManager();
+            if (am != null && am.isTouchExplorationEnabled()) {
+                ret = true;
+            }
         }
-        return ret;
-    }
 
-    /**
-     * If a listener is previously registered, unregisters then registers this one
-     * @param am The accessibility manager to unregister from/register to
-     * @param listener The new listener
-     * @return True if successful
-     */
-    private boolean registerTouchExploreChangeListener(@NonNull AccessibilityManager am, @NonNull AccessibilityManager.TouchExplorationStateChangeListener listener) {
-        boolean ret = false;
-        if (unregisterTouchExploreChangeListener(am) && am.addTouchExplorationStateChangeListener(listener)) {
-            touchExplorationStateChangeListener = listener;
-            ret = true;
-        }
         return ret;
     }
 
-    protected boolean setTouchExploreChangeListener(@Nullable AccessibilityManager.TouchExplorationStateChangeListener listener) {
-        boolean ret = false;
+    @TargetApi(Build.VERSION_CODES.KITKAT)
+    protected void setTouchExploreChangeListener(@Nullable AccessibilityManager.TouchExplorationStateChangeListener listener) {
         if (listener != touchExplorationStateChangeListener) {
             AccessibilityManager am = getAccessibilityManager();
             if (am != null) {
-                if (listener != null) {
-                    ret = registerTouchExploreChangeListener(am, listener);
+
+                if (touchExplorationStateChangeListener != null) {
+                    am.removeTouchExplorationStateChangeListener((AccessibilityManager.TouchExplorationStateChangeListener) touchExplorationStateChangeListener);
                 }
-                else {
-                    ret = unregisterTouchExploreChangeListener(am);
+
+                if (listener != null) {
```

Comment:
Sure. I’ll put one in.

On Tue, Aug 2, 2016 at 10:54 AM, Christopher Lee <notifications@github.com>
wrote:

> In VideoLocker/src/main/java/org/edx/mobile/player/PlayerFragment.java
> <https://github.com/edx/edx-app-android/pull/787#discussion_r73170688>:
>
> >                  }
> > -                else {
> > -                    ret = unregisterTouchExploreChangeListener(am);
> > +
> > +                if (listener != null) {
>
> I think a comment would be helpful here.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/edx/edx-app-android/pull/787/files/c1d3cc4578cdf9f6cf04ac7b03c6ee26fc7ae722#r73170688>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AR10T7oWM8pD7LNCpb-aO71L_oP9OEC6ks5qb1o2gaJpZM4JZ9Gg>
> .
>



-- 
Regards,

Marc DiNino
Android Engineer
EDX